subsys/net/lib/nrf_cloud/src/nrf_cloud_fota.c

```diff
@@ -734,6 +780,11 @@ static void http_fota_handler(const struct fota_download_evt *evt)
 
 	if (ret) {
 		LOG_ERR("Failed to send job update to cloud: %d", ret);
+		/* The done event is normally sent on the MQTT ACK, but if the
+		 * status update fails there will be no ACK. Send the done event
+		 * so the device can proceed to install the FOTA update.
+		 */
+		send_fota_done_event();
```

Comment:
I think this could technically trigger during `FOTA_DOWNLOAD_EVT_PROGRESS`, right? Would that cause a reboot mid-way through a download if it were to occur?

Reply:
no, see contents of `send_fota_done_event()`.   the `NRF_CLOUD_FOTA_IN_PROGRESS` state is set after the download is finished.

Reply:
Ah, I see now!

I think it might be easier to follow if those conditions are outside the function call, rather than inside -- What is the reason for them to be inside?

Alternatively, extend the name of the function:

`send_fota_done_event_if_download_finished()`

Reply:
how about: 
`send_done_event_if_necessary_conditions_are_met()`

Reply:
I like that, but maybe just

`send_done_event_if_conditions_met`

Reply:
I see that you switched to `if_done`, that works too!